drop-folders

konstantin-stepanets opened the pull request:
### PR information
**What kind of change does this PR introduce?** (check one with "x")
- [] Bugfix
- [x] Feature
- [] Code style update (formatting, local variables)
- [] Refactoring (no functional changes, no api changes)
- [] Build related changes
- [] CI related changes
- [] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
- [] Yes
- [x] No


If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

---

### Other information

**Where should the reviewer start?**

**How should this be manually tested?**

**Any background context you want to provide?**

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kaltura/kmc-ng/238)
<!-- Reviewable:end -->

konstantin-stepanets commented:
Review status: 0 of 28 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 124 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KxE0wWgJxI5kaI-X0bW:-KxE0wWgJxI5kaI-X0bX:bkcjb6m) ([raw file](https://github.com/kaltura/kmc-ng/blob/56f1a5b3d16dfae02c42b4c9c79ddd76f21dfe5c/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L124)):*
> response.objects.forEach(object => {
<details><summary><i alt="summary">Quoted 18 lines of code…</i></summary>
>               if(object instanceof KalturaDropFolder) {
>                 df = object;
>                 if(JSON.stringify(df.fileHandlerType) == JSON.stringify(KalturaDropFolderFileHandlerType.content)) {
>                   let cfg: KalturaDropFolderContentFileHandlerConfig = df.fileHandlerConfig as KalturaDropFolderContentFileHandlerConfig;
>                   if (_flags && DropFolderListType.ADD_NEW && cfg.contentMatchPolicy === KalturaDropFolderContentFileHandlerMatchPolicy.addAsNew) {
>                     this.ar.push(df);
>                   }
>                   else if (_flags & DropFolderListType.MATCH_OR_KEEP && cfg.contentMatchPolicy === KalturaDropFolderContentFileHandlerMatchPolicy.matchExistingOrKeepInFolder) {
>                     this.ar.push(df);
>                   }
>                   else if (_flags & DropFolderListType.MATCH_OR_NEW && cfg.contentMatchPolicy === KalturaDropFolderContentFileHandlerMatchPolicy.matchExistingOrAddAsNew) {
>                     this.ar.push(df);
>                   }
>                 } else if (_flags & DropFolderListType.XML_FOLDER && df.fileHandlerType === KalturaDropFolderFileHandlerType.xml) {
>                   this.ar.push(df);
>                 }
>               }
>             });
</details>
+@amirch1  Could you plz review this part of code and compare with PRD? I'm not sure that some stuff was "converted" from action script to JS correctly (some properties don't exist).  Some related places were temporary commented untill it's fixed (e.g. Dropfolder name column at the table)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/238)*
<!-- Sent from Reviewable.io -->

konstantin-stepanets commented:
Review status: 0 of 28 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 124 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KxE0wWgJxI5kaI-X0bW:-KxE2x94BOPn2LiCfFYb:b-wirko1) ([raw file](https://github.com/kaltura/kmc-ng/blob/56f1a5b3d16dfae02c42b4c9c79ddd76f21dfe5c/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L124)):*
<details><summary><i>Previously, konstantin-stepanets (Konstantin Stepanets) wrote…</i></summary><blockquote>

> response.objects.forEach(object => {
>               if(object instanceof KalturaDropFolder) {
>                 df = object;
>                 if(JSON.stringify(df.fileHandlerType) == JSON.stringify(KalturaDropFolderFileHandlerType.content)) {
>                   let cfg: KalturaDropFolderContentFileHandlerConfig = df.fileHandlerConfig as KalturaDropFolderContentFileHandlerConfig;
>                   if (_flags && DropFolderListType.ADD_NEW && cfg.contentMatchPolicy === KalturaDropFolderContentFileHandlerMatchPolicy.addAsNew) {
>                     this.ar.push(df);
>                   }
>                   else if (_flags & DropFolderListType.MATCH_OR_KEEP && cfg.contentMatchPolicy === KalturaDropFolderContentFileHandlerMatchPolicy.matchExistingOrKeepInFolder) {
>                     this.ar.push(df);
>                   }
>                   else if (_flags & DropFolderListType.MATCH_OR_NEW && cfg.contentMatchPolicy === KalturaDropFolderContentFileHandlerMatchPolicy.matchExistingOrAddAsNew) {
>                     this.ar.push(df);
>                   }
>                 } else if (_flags & DropFolderListType.XML_FOLDER && df.fileHandlerType === KalturaDropFolderFileHandlerType.xml) {
>                   this.ar.push(df);
>                 }
>               }
>             });
+@amirch1  Could you plz review this part of code and compare with PRD? I'm not sure that some stuff was "converted" from action script to JS correctly (some properties don't exist).  Some related places were temporary commented untill it's fixed (e.g. Dropfolder name column at the table)
</blockquote></details>

The question was copied again because of mess above, sorry.
+@amirch1 Could you plz review this part of code and compare with PRD? I'm not sure that some stuff was "converted" from action script to JS correctly (some properties don't exist). Some related places were temporary commented untill it's fixed (e.g. Dropfolder name column at the table)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/238)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 17 of 28 files at r1, 8 of 11 files at r3.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---

*[src/app-config/index.ts, line 175 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykKjmCpzF8D7_vCKBb:-KykKjmDdi_Vn_li99l0:bdpeos6) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/app-config/index.ts#L175)):*
> ```XML
>         "dropFolders" : {
>           "createdAtDateRange" : "2005:2030",
>           "bulkActionsLimit": 2
> ```

change it to 50

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html, line 1 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykL836xyFG4KhThgR9:-KykL836xyFG4KhThgRA:b-6cbx90) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html#L1)):*
> ```HTML
> <k-area-blocker kFillHeight [showLoader]="(_dropFoldersService.state$ | async)?.loading">
> ```

is it using the new scroll system?

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html, line 31 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykVDDBcVXwPP0XzoR9:-KykVDDBcVXwPP0XzoRA:b-rz5kx3) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html#L31)):*
> ```HTML
>                   [sortable]="true">
>           <ng-template let-col let-folder="rowData" pTemplate="body">
>             <span [kTooltip]="dateTooltip(folder)">{{ folder[col.field] | date: 'dd/MM/yy' }}</span>
> ```

shouldn't we use the `kDate` here?

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 13 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykYBcLKbb26o0x47Sr:-KykYBcLKbb26o0x47Ss:b-2fjonl) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L13)):*
> ```XML
>   templateUrl: './drop-folders-list-table.component.html',
>   styleUrls: ['./drop-folders-list-table.component.scss'],
>   providers: [DatePipe]
> ```

you don't need date pipe to manipulate dates. please use `moment` instead

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 43 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykViU1eYPsByCrUDbS:-KykViU1eYPsByCrUDbT:b-voox7z) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L43)):*
> ```XML
>     private _appLocalization: AppLocalization,
>     private cdRef: ChangeDetectorRef,
>     private _datePipe: DatePipe,
> ```

see my comment on the `providers`

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 93 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykWjpVeP51mfl_g6BF:-KykWjpVeP51mfl_g6BG:b-awsqmr) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L93)):*
> ```XML
> 
>   scrollToTop() {
>     const scrollBodyArr = this.dataTable.el.nativeElement.getElementsByClassName("ui-datatable-scrollable-body");
> ```

this is not needed with the new scroll system. please use browser service instead. It should expose a relevant method.   Please consult with +@amirch1

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 101 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykWt29NWwP8buId6qP:-KykWt29NWwP8buId6qQ:byyjqc9) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L101)):*
> ```XML
> 
>   ngAfterViewInit() {
>     const scrollBody = this.dataTable.el.nativeElement.getElementsByClassName("ui-datatable-scrollable-body");
> ```

same here, we don't use this approach anymore, I think we have a dedicated directive for that. Please consult with +@amirch1

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html, line 50 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kyk_YeZU-0m2q_d1Dkj:-Kyk_YeZU-0m2q_d1Dkk:bz3qmbw) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html#L50)):*
> ```HTML
>         class="kDropFoldersTable"
>         [dropFolders]="(_dropFoldersService.dropFolders$ | async)?.items"
>         (isEntryExist)="checkIfEntryExist($event)"
> ```

Both the name of the output parameter and the name of the method are misleading. It doesn't check if entry exists, it navigate to that entry if exist. 

Another thing, are there situations when the server returns an entry id yet it doesn't exists in the server? why not assuming you can navigate to that entry?

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html, line 52 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kyk_1QF0YC6fEJqdgMm:-Kyk_1QF0YC6fEJqdgMn:buih5xp) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html#L52)):*
> ```HTML
>         (isEntryExist)="checkIfEntryExist($event)"
>         (deleteDropFolderFiles)="deleteDropFolderFiles($event)"
>         (onSelectedDropFoldersChange)="_selectedDropFoldersChange($event)"
> ```

you can use here the two way binding syntax 

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.scss, line 262 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykXJs9TQYoLTmA5U0p:-KykXJs9TQYoLTmA5U0q:bwyvukr) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.scss#L262)):*
> ```SCSS
> }
> 
> /deep/ .kActionsMenu .ui-menuitem-text {
> ```

+@amirch1 please advice

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 27 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykXOlDcEzSefj7JZfM:-KykXOlDcEzSefj7JZfN:bwyvukr) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L27)):*
> ```XML
>   templateUrl: './drop-folders-list.component.html',
>   styleUrls: ['./drop-folders-list.component.scss'],
>   providers: [BulkDeleteService, FolderFileStatusPipe]
> ```

+@amirch1 please advice

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 32 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykYMujZZrStxd4amBi:-KykYMujZZrStxd4amBj:bspd4vm) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L32)):*
> ```XML
> export class DropFoldersListComponent implements OnInit, OnDestroy {
>   @ViewChild(DropFoldersListTableComponent) private dataTable: DropFoldersListTableComponent;
>   @ViewChild('statusFilter') private statusFilter: StatusesFilterComponent;
> ```

why not using the type as identifier? why using a string instead?

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 35 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykYRKiyrl19zM_nkOY:-KykYRKiyrl19zM_nkOZ:b-rnqzx1) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L35)):*
> ```XML
>   @ViewChild('statusesFilterPopup') public statusesFilterPopup: PopupWidgetComponent;
> 
>   _isBusy = false;
> ```

should use the new support we added to `tag('block-shell')` operator. It is already in master. 

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 134 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykaGMa-5rk_eNj-qmA:-KykaGMa-5rk_eNj-qmB:b-egj9sf) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L134)):*
> ```XML
>   }
> 
>   updateFilters(filter: Filter, flag?: number) { // if flag == 1 we won't push filter to activeFilters
> ```

consider replacing the flag with something more relevant like `updateActiveFilters: boolean` 

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 229 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykarJb0SzrYErFFtu2:-KykarJb0SzrYErFFtu3:b-rnqzx1) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L229)):*
> ```XML
> 
>   private deleteDropFiles(ids:number[]): void {
>     this._isBusy = false;
> ```

should use the new support we added to `tag('block-shell')` operator. It is already in master. 

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 243 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykamUwle272MGh4evQ:-KykamUwle272MGh4evR:bx6315c) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L243)):*
> ```XML
>               message: this._appLocalization.get('applications.content.dropFolders.errors.errorDropFoldersFiles'),
>               buttons: [{
>                 label: this._appLocalization.get('app.common.ok'),
> ```

please provide a refresh action instead of just dismissing the error

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 255 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykb0pLd3V7D1C-_2nN:-Kykb0pLd3V7D1C-_2nO:bb94wau) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L255)):*
> ```XML
> 
>     if(ids.length > environment.modules.contentEntries.bulkActionsLimit) {
>       this._browserService.confirm(
> ```

is this taken from the PRD? please provide a link

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 124 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KxE0wWgJxI5kaI-X0bW:-KykbmbAbMnh0WgbXAIM:b-rtg0gd) ([raw file](https://github.com/kaltura/kmc-ng/blob/56f1a5b3d16dfae02c42b4c9c79ddd76f21dfe5c/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L124)):*
<details><summary><i>Previously, konstantin-stepanets (Konstantin Stepanets) wrote…</i></summary><blockquote>

The question was copied again because of mess above, sorry.
+@amirch1 Could you plz review this part of code and compare with PRD? I'm not sure that some stuff was "converted" from action script to JS correctly (some properties don't exist). Some related places were temporary commented untill it's fixed (e.g. Dropfolder name column at the table)
</blockquote></details>

+@amirch1 please review this

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 50 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykbMO5hT3kXTTThdbK:-KykbMO5hT3kXTTThdbL:b-h21w9u) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L50)):*
> ```XML
> 	});
> 	ar: any[] = [];
> 	allStatusesList: string = KalturaDropFolderFileStatus.downloading + "," +
> ```

do you have to provide the complete list, is this what happens in the legacy?

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 117 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykbstjy-j3TRRa3Fx2:-Kykbstjy-j3TRRa3Fx3:bzi390x) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L117)):*
> ```XML
> 			.subscribe(
> 				response => {
> 					if (response.objects.length) {
> ```

you are missing else part to perform initialization/cleanups of service state

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 122 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykbyBOt5OTwAU-d-yR:-KykbyBOt5OTwAU-d-yS:b-52rmlx) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L122)):*
> ```XML
> 						this.ar = [];
> 						response.objects.forEach(object => {
> 							if (object instanceof KalturaDropFolder) {
> ```

you are missing else part to perform initialization/cleanups of service state or to provide an error

---

*[src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.html, line 1 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykco-BBot5TtI1Mb2C:-Kykco-BBot5TtI1Mb2D:bki4hnb) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.html#L1)):*
> ```HTML
> <div class="kFiltersGroupTitle">
> ```

don't we have such a component in a shared location? if not we should consider having such a component. Lets discuss

---

*[src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.scss, line 32 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykcvkIj4qRkLDt4mme:-KykcvkIj4qRkLDt4mmf:b-w5qfgw) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.scss#L32)):*
> ```SCSS
> }
> 
> :host /deep/ .kDateRange .kDateRangeFirst .ui-calendar.ui-calendar-w-btn {
> ```

you can unify all the `:host /deep/` together since scss support nesting declaration.

---

*[src/applications/content-drop-folders-app/drop-folders-list/pipes/folder-name.pipe.ts, line 6 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykdg8l1LgDMy6kVMTM:-Kykdg8l1LgDMy6kVMTN:bwc6j7u) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/pipes/folder-name.pipe.ts#L6)):*
> ```XML
> 
> @Pipe({ name: 'kFolderName' })
> export class FolderNamePipe implements PipeTransform {
> ```

did you consider preparing the data in the service instead of having this pipe and the other one that is in use?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/238)*
<!-- Sent from Reviewable.io -->

konstantin-stepanets commented:
Review status: 14 of 27 files reviewed at latest revision, 24 unresolved discussions.

---

*[src/app-config/index.ts, line 175 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykKjmCpzF8D7_vCKBb:-Kz3A17OJmO4akz87zz9:b32w0x) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/app-config/index.ts#L175)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

change it to 50
</blockquote></details>

OK

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html, line 1 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykL836xyFG4KhThgR9:-Kz4S1jmBqcSG-F7_zle:b-lqalos) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html#L1)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

is it using the new scroll system?
</blockquote></details>

Updated, should use now.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html, line 31 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykVDDBcVXwPP0XzoR9:-Kz4TAjlDvzrRRH1l3i8:bcmd0ml) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.html#L31)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

shouldn't we use the `kDate` here?
</blockquote></details>

I thought so too but if to be very meticulous then according to the png in attachments to the PRD (https://kaltura.atlassian.net/wiki/spaces/KMCng/pages/234162917/Drop+Folders?preview=/234162917/234166711/image2017-10-10_10-33-55.png) it was without HH:mm, therefore I decided to use the custom.
But np, changed to kDate pipe.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 13 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykYBcLKbb26o0x47Sr:-Kz8Oj87hsGQsYRD7U9n:b2nhktp) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L13)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you don't need date pipe to manipulate dates. please use `moment` instead
</blockquote></details>

OK.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 43 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykViU1eYPsByCrUDbS:-Kz8PWJqD9l3Kzrk8TLD:b32w0x) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L43)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

see my comment on the `providers`
</blockquote></details>

OK

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 93 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykWjpVeP51mfl_g6BF:-Kz4X4iznJ0aiI41676_:bduri3z) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L93)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

this is not needed with the new scroll system. please use browser service instead. It should expose a relevant method.   Please consult with +@amirch1
</blockquote></details>

Ok, fixed.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts, line 101 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykWt29NWwP8buId6qP:-Kz4_aYC8tfPa_oNCJEL:bexs1fq) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list-table.component.ts#L101)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

same here, we don't use this approach anymore, I think we have a dedicated directive for that. Please consult with +@amirch1
</blockquote></details>

Ok. Have found and fixed.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html, line 50 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kyk_YeZU-0m2q_d1Dkj:-Kz8d5oxgJOoXqpxpq-H:b-5ypnar) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html#L50)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

Both the name of the output parameter and the name of the method are misleading. It doesn't check if entry exists, it navigate to that entry if exist. 

Another thing, are there situations when the server returns an entry id yet it doesn't exists in the server? why not assuming you can navigate to that entry?
</blockquote></details>

Renamed. Regarding an entry id that doesn't exist, server will return an appropriate error and it will be shown.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 32 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykYMujZZrStxd4amBi:-Kz9TefGp07lc-x6zmM6:bgmcf3n) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L32)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

why not using the type as identifier? why using a string instead?
</blockquote></details>

I didn't think that there is a very big difference and it could be critical.. Updated.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 35 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykYRKiyrl19zM_nkOY:-Kz9dGsMEIT9muLF-_Be:baey9z6) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L35)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

should use the new support we added to `tag('block-shell')` operator. It is already in master. 
</blockquote></details>

Np, updated, but to be honest this isn't very nice directly for this case (imho) - https://www.screencast.com/t/kRpHBblC5kr

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 134 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykaGMa-5rk_eNj-qmA:-Kz9wj-i5U7b2QKx1F_B:b32w0x) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L134)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

consider replacing the flag with something more relevant like `updateActiveFilters: boolean` 
</blockquote></details>

OK

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 229 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykarJb0SzrYErFFtu2:-Kz9x5KQHB2sqyAxmSnK:b32w0x) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L229)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

should use the new support we added to `tag('block-shell')` operator. It is already in master. 
</blockquote></details>

OK

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 255 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykb0pLd3V7D1C-_2nN:-Kz9xSZMNDzKtdiqMbxA:b-k2tgrn) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L255)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

is this taken from the PRD? please provide a link
</blockquote></details>

No, but as I remember Amir said that this is applicable for all such deletions. Was fixed to environment.modules.dropFolders.bulkActionsLimit.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 124 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KxE0wWgJxI5kaI-X0bW:-Kz9xvylCwSr_upCsJFz:b-jtyxc8) ([raw file](https://github.com/kaltura/kmc-ng/blob/56f1a5b3d16dfae02c42b4c9c79ddd76f21dfe5c/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L124)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

+@amirch1 please review this
</blockquote></details>

This part has already been reviewed and fixed by Amir.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 50 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykbMO5hT3kXTTThdbK:-Kz9ybX-It8c3rRKe3PM:b-2ew03k) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L50)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

do you have to provide the complete list, is this what happens in the legacy?
</blockquote></details>

I re-used legacy code from the PRD (https://www.screencast.com/t/OUTUZU5j25o), but as I know we have more statuses.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/238)*
<!-- Sent from Reviewable.io -->

konstantin-stepanets commented:
Review status: 14 of 27 files reviewed at latest revision, 24 unresolved discussions.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html, line 52 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kyk_1QF0YC6fEJqdgMm:-KzTtS4JiriL2e9YzMwd:bdoti5r) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.html#L52)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you can use here the two way binding syntax 
</blockquote></details>

Could you plz show an example how it should be? (we have a dozen or two of the tables), I'm a bit stuck at this point.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts, line 243 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykamUwle272MGh4evQ:-KzNsRlovJEKdEuGgTH1:bslbf0x) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders-list.component.ts#L243)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

please provide a refresh action instead of just dismissing the error
</blockquote></details>

OK. *I think you mean the "retry" and "cancel" actions.

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 117 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykbstjy-j3TRRa3Fx2:-KzOQBsM7qiOvrIEOzWo:bq3yp9o) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L117)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you are missing else part to perform initialization/cleanups of service state
</blockquote></details>

I have the _loadDropFoldersList method that is called at OnInit (within reload()) and in successful case in its turn called the loadDropFoldersFiles (that returns a list of entries) . How to apply correctly the new .tag('block-shell') logic? If I'll add this tag to both methods or at least for the first one, I'll get an error - https://www.screencast.com/t/HS7r7SBszBG. This is what I mean if we are talking about state management and if to speak about UI (spinner behaviors for such cases).
Or you mean that it's enought just to show an error? (throw new Error("error occurred in action \'_loadDropFoldersList\'"); )

---

*[src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts, line 122 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykbyBOt5OTwAU-d-yR:-KzOn7DHx5qiiPT8LMut:bzcpawp) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/drop-folders.service.ts#L122)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you are missing else part to perform initialization/cleanups of service state or to provide an error
</blockquote></details>

You are right, we have a lot of such similar places without else part. Am I right that it's enough to just add an error like:
throw new Error(`invalid type provided, expected KalturaDropFolder, got ${typeof response}`) ?

---

*[src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.html, line 1 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykco-BBot5TtI1Mb2C:-KzPBjX2Cc-iDKlP-Ttv:b-hlgb5s) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.html#L1)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

don't we have such a component in a shared location? if not we should consider having such a component. Lets discuss
</blockquote></details>

No, we don't have. Have moved my component to shared directory but used it only for drop folders at this moment. If you'll approve, we'll be able to investigate and update the other similar places.

---

*[src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.scss, line 32 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-KykcvkIj4qRkLDt4mme:-KzPF4h7s6gdPMrAplCr:b32w0x) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/dates-filters/dates-filters.component.scss#L32)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you can unify all the `:host /deep/` together since scss support nesting declaration.
</blockquote></details>

OK

---

*[src/applications/content-drop-folders-app/drop-folders-list/pipes/folder-name.pipe.ts, line 6 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/238#-Kykdg8l1LgDMy6kVMTM:-KzT9sdd8IS1kgGePVik:b-vv5zcu) ([raw file](https://github.com/kaltura/kmc-ng/blob/5a8e4e86b2250f2ddf69d7b206b9ebd89adb57f2/src/applications/content-drop-folders-app/drop-folders-list/pipes/folder-name.pipe.ts#L6)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

did you consider preparing the data in the service instead of having this pipe and the other one that is in use?
</blockquote></details>

Was removed the "FolderNamePipe" and prepared such data in the service. The second one - "FolderFileStatusPipe", seems contains a lot of logic for service, decided to left in the separate pipe.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/238)*
<!-- Sent from Reviewable.io -->
